transform-plugins/src/main/java/co/cask/hydrator/plugin/ValueMapper.java

```diff
@@ -0,0 +1,228 @@
+                  Schema.of(Schema.Type.STRING)));
+        } else {
+          outputFields.add(inputField);
+        }
+      }
+    }
+    outputSchema = Schema.recordOf(inputSchema.getRecordName() + ".formatted", outputFields);
+    schemaCache.put(inputSchema, outputSchema);
+    return outputSchema;
+  }
+
+  /**
+   * Parse configuration provided by user
+   */
+  public void parseConfiguration(String mapping, TransformContext context) {
+    LOG.debug("in parseConfiguration:" + mapping);
+    JsonParser jsonParser = new JsonParser();
+    JsonElement jsonMapping = jsonParser.parse(mapping);
+    LOG.debug("in parseConfiguration:" + jsonMapping.toString());
+    mappingArray = jsonMapping.getAsJsonObject().get("mapping").getAsJsonArray();
+    for (JsonElement mappingElement : mappingArray) {
+      String sourceField = mappingElement.getAsJsonObject().get("mapperField").getAsString();
+      String targetField = mappingElement.getAsJsonObject().get("mappingField").getAsString();
+      fieldsMapping.put(sourceField, targetField);
+      ValueMapperLookUp lookUpTable = init(context, mappingElement.getAsJsonObject().get("lookup").toString());
+      datasetMapping.put(sourceField, lookUpTable);
+    }
+  }
+
+  /**
+   * Creates a look up table
+   */
+  private ValueMapperLookUp init(LookupProvider context, String lookUp) throws IllegalArgumentException {
+
+    LOG.debug("in init:" + lookUp);
+    LookupConfig lookUpConfig;
+    try {
+      lookUpConfig = GSON.fromJson(lookUp, LookupConfig.class);
+    } catch (JsonSyntaxException e) {
+      throw new IllegalArgumentException("Invalid lookup config. Expected map of string to string", e);
+    }
+
+    ValueMapperLookupProvider valueMapperLookupProvider = new ValueMapperLookupProvider(context, lookUpConfig);
+    String lookUpTable = parseLookUpTableName(lookUp);
+
+    ValueMapperLookUp lookupTable = valueMapperLookupProvider.provide(lookUpTable);
+    return lookupTable;
+  }
+
+  /**
+   * parse lookup table name from lookup object
+   */
+  private String parseLookUpTableName(String lookUp) {
+    String tableName = null;
+    JsonParser jsonParser = new JsonParser();
+    JsonElement jsonMapping = jsonParser.parse(lookUp);
+    JsonElement table = jsonMapping.getAsJsonObject().get("tables");
+    for (Map.Entry entry : table.getAsJsonObject().entrySet()) {
+      tableName = entry.getKey().toString();
+      break;
+    }
+    return tableName;
+  }
+
+  @Override
+  public void initialize(TransformContext context) throws Exception {
+    super.initialize(context);
+    parseConfiguration(config.mapping, context);
+  }
+
+  /**
+   * Configuration for the ValueMapper transform.
+   */
+  public static class Config extends PluginConfig {
+
+    //TODO input parameters may be changed after having three different input params from UI
+    @Name("mapping")
+    @Description("Specify the source and target field mapping and lookup dataset name.")
```

Comment:
the 'tables' section is a json object, but only one entry in it is used and only the key is used. Which entry is used is dependent on the Map implementation that is used when deserializing this.  I'm assuming this will be revisited based on the comment above, but there are many things wrong with this.